uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java

```diff
@@ -333,6 +334,31 @@ public void connect(String clientDeviceId, String componentId, String brokerId,
         throw lastException;
     }
 
+    /**
+     * Creates MQTT connection.
+     *
+     * @param clientDeviceId the id of the device (thing name) as defined by user in scenario
+     * @param componentId  the componentId of MQTT client
+     * @param brokerId the id of broker, before must be discovered or added by default
+     * @param mqttVersion the MQTT version string
+     * @throws RuntimeException throws in fail case
+     *
+     */
+    @SuppressWarnings({"PMD.UseObjectForClearerAPI", "PMD.AvoidCatchingGenericException"})
+    @And("I can not connect device {string} on {word} to {string} using mqtt {string}")
+    public void canNotConnect(String clientDeviceId, String componentId, String brokerId, String mqttVersion) {
+        try {
+            Mqtt5ConnAck connAck = connect(clientDeviceId, componentId, brokerId, mqttVersion);
+            int actualStatus = connAck.getReasonCode();
+
+            if (actualStatus == 0 || actualStatus == 1) {
```

Comment:
1. I think it is invalid check. reason code 1 is not a success for CONNACK. Please check reason code values https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901031

2. Constants should be a part of enum, for example create ConnectReasonCode.java in control near by PublishReasonCode.java/SubscribeReasonCode.java

Reply:
I deleted check logic in f59cbc207e592c1b5558492365897fe2bdcfbcb3